src/main/java/com/fasterxml/jackson/core/StreamReadConstraints.java

```diff
@@ -42,6 +42,15 @@
      */
     public static final int DEFAULT_MAX_STRING_LEN = 5_000_000;
 
+    /**
+     * Limit for the maximum magnitude of Scale of {@link java.math.BigDecimal} that can be
+     * converted to {@link java.math.BigInteger}; actual limit value is either
+     * {@code 10 * maxNumLen} or this value, whichever is smaller.
```

Comment:
There doesn't seem to be any code implementing the `10 * maxNumLen` logic. Since maxNumLen defaults to 1000 - the effective maxScale is 10,000 under this logic.

I prefer the simpler solution of having a global maxScale independent of the maxNumLen.

Reply:
Ah yes, thanks, need to remove comment -- it was my plan A before coming to conclusion simplicity rules.